src/lib/dereference.js

```diff
@@ -7,28 +8,42 @@ export default function dereference(
   dataSources,
   config = {deleteKeys: false}
 ) {
-  const replacer = (key, parent) => {
+  const replacer = (key, parent, srcPath) => {
     if (!SRC_ATTR_PATTERN.test(key)) {
       return;
     }
 
-    const srcRef = parent[key];
-    const data = dataSources[srcRef];
     const dataKey = key.replace(SRC_ATTR_PATTERN, '');
+    const traceType = parent.type;
 
-    if (config.deleteKeys && !(srcRef in dataSources)) {
-      delete parent[dataKey];
-      return;
+    // making this into an array to more easily lookup 1d and 2d srcs in dataSourceOptions
```

Comment:
it's easier to have this in an array format to lookup srcs in dataOptions, this behaviour here was broken in current dereferencing function for array srcs